### DISCUSSION TITLE
Android not receiving video packets

### DISCUSSION THREAD

**Discussion by frogblueDFi** [juha-h/baresip-studio#394] [Q&A]
Hi there!

I am still a newbie to baresip but have been successful in implementing a macOS/iOS SIP app based on baresip.
Audio and video is working perfect both via direct calls and via our own SIP server.

Moving to Android (using Qt as a cross platform framework), things are quite easy after correctly cross compiling modules.
Audio is working fine both for direct calls and via server.

But: video is only working via direct calls, NOT via the SIP server.
No video packets are delivered from the baresip registered baresip module as when using direct calls. 

Checked already switching transport, played with stun, ice and turn modules.
Seems not be related to the SIP server, as e.g Linphone shows video with the same account.

Guess it must be related to some Android permission or whatever, but also checked already manifest file and permissions as android.permission.INTERNET etc.
Accounts, configurations as stun etc. are exactly the same as for macOS/iOS.

Would appreciate any(!) hints where to go into deep as I am out of ideas for the moment. Maybe someone out there had similar issues.

Man many thanx in advance
Dietmar

**Comment by juha-h** [comment]
I don't think it is related to any Android permissions.  It was not clear to me if any video packets are received or send by baresip.   Check if video stream IP addresses are OK in SDP of INVITE/200 response and if you get any error messages to logcat.

**Comment by frogblueDFi** [comment]
Solved: After having implemented the video capturing module on Android (before I had only receiving one-way on Android), video is functional in both directions via our SIP server.
Tried again/counter checked without loading the capturing "vidsrc" module: no video.
Seems that when negotiating call features, video in both directions must be available.
But why is this different from direct call? Baresip specific or related to our SIP server configuration?

Thanx for your support!

**Comment by juha-h** [comment]
frogblueDFi writes:

> Seems that when negotiating call features, video in both directions
> must be available.

Yes, it is not currently possible to select when call is made or answered
that video is sendonly or recvonly.  Would complicate the user interface
too much.

> But why is this different from direct call? Baresip specific or
> related to our SIP server configuration?

baresio app does not behave differently with direct calls or calls via a
SIP proxy.
